torch/onnx/_internal/exporter.py

```diff
@@ -184,6 +216,70 @@ def resolve(value: Optional[T], fallback: Union[T, Callable[[], T]]) -> T:
                     assert hasattr(self, key), f"Unresolved option '{key}'"
 
 
+@contextlib.contextmanager
+def enable_fake_mode():
+    """Enable fake mode for the duration of the context.
+
+    Internally it instantiates a `FakeTensorMode` context manager that converts
+    user input and model parameters into `FakeTensor`.
+
+    A [FakeTensor](https://github.com/pytorch/pytorch/blob/main/torch/_subclasses/fake_tensor.py#L870)
+    is a `torch.Tensor` with the ability to run PyTorch code without having to
+    actually do computation through tensors allocated on a `meta` device. Because
+    there is no actual data being allocated on the device, this API allows for
+    exporting large models without the actual memory footprint needed for executing it.
+
+    It is highly recommended to enable fake mode when exporting models that
+    are too large to fit into memory.
+
+    Returns:
+        A `ONNXFakeContext` object that must be passed to `torch.onnx.dynamo_export`
+        through the `ExportOptions.fake_context` argument.
+
+    Example::
+
+        # xdoctest: +REQUIRES(env:TORCH_DOCTEST_ONNX)
+        >>> import torch
+        >>> import torch.onnx
+        >>> class MyModel(torch.nn.Module):  # Dummy model
+        ...     def __init__(self) -> None:
+        ...         super().__init__()
+        ...         self.linear = torch.nn.Linear(2, 2)
+        ...     def forward(self, x):
+        ...         out = self.linear(x)
+        ...         return out
+        >>> with torch.onnx.enable_fake_mode() as fake_context:
+        ...     my_nn_module = MyModel()
+        ...     arg1 = torch.randn(2, 2, 2)  # positional input 1
+        >>> export_options = torch.onnx.ExportOptions(fake_context=fake_context)
+        >>> export_output = torch.onnx.dynamo_export(
+        ...     my_nn_module,
+        ...     arg1,
+        ...     export_options=export_options
+        ... )
+        >>> # Saving model WITHOUT initializers
+        >>> export_output.save("my_model_without_initializers.onnx")
+        >>> # Saving model WITH initializers
+        >>> export_output.save("my_model_with_initializers.onnx", model_state_dict=MyModel().state_dict())
+
+    .. warning::
+        This API is experimental and is *NOT* backward-compatible.
+
+    """
+    from torch._subclasses import fake_tensor
+    from torch.fx.experimental.symbolic_shapes import ShapeEnv
+
+    fake_mode = fake_tensor.FakeTensorMode(
+        allow_non_fake_inputs=False,
+        shape_env=ShapeEnv(
+            allow_scalar_outputs=False, allow_dynamic_output_shape_ops=False
```

Comment:
Forgot to comment on this. This overrides internal fake tensor/dynamo settings that other export api may set. Best call out with comments.

Reply:
done